### PR TITLE
Scala 3.0.0-RC2

### PR DESCRIPTION
**Pull request by sideeffffect**

### CODE REVIEW

**Comment by sideeffffect** [comment]
:thinking: 
```scala
[error] -- Error: /home/runner/work/zio/zio/test/shared/src/main/scala/zio/test/DefaultMutableRunnableSpec.scala:22:6 
[error] 22 |class DefaultMutableRunnableSpec extends MutableRunnableSpec(ZLayer.identity[TestEnvironment], TestAspect.identity)
[error]    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |class DefaultMutableRunnableSpec cannot be instantiated since it has conflicting base type arguments zio.test.Annotations & zio.test.environment.Live & zio.test.Sized & 
[error]    |  zio.test.environment.TestClock
[error]    | & zio.test.TestConfig & zio.test.environment.TestConsole & 
[error]    |  zio.test.environment.TestRandom
[error]    | & zio.test.environment.TestSystem and zio.ZEnv
```

**Comment by adamgfraser** [comment]
@sideeffffect It looks like adding a type alias for the intersection types resolves the issue. I submitted a PR against your branch with that change. I will open an issue with the Scala 3 team but that should get us going for now.

**Comment by adamgfraser** [comment]
Commented on lampepfl/dotty#11995, which I believe is the same issue.

**Comment by sideeffffect** [comment]
Thanks Adam! :heart:

**Comment by sideeffffect** [comment]
Dotty build seemed to have failed only on a flaky test :smile: Could somebody restart it, please?

File: test-tests/shared/src/test/scala/zio/test/MutableRunnableSpecSpec.scala [hunk 1]
```diff
@@ -3,10 +3,10 @@ package zio.test
 import zio.test.Assertion._
 import zio.test.TestAspect._
 import zio.test.environment.TestEnvironment
-import zio.{Ref, ZIO}
+import zio.{Has, Ref, ZIO}
 
 object MutableRunnableSpecSpec
-    extends MutableRunnableSpec(
+    extends MutableRunnableSpec[MutableRunnableSpecSpecCompat.Environment](
       TestEnvironment.any ++ Ref.make(0).toLayer,
```
Comment (adamgfraser) [inline test-tests/shared/src/test/scala/zio/test/MutableRunnableSpecSpec.scala:10]:
```suggestion
      TestEnvironment.any ++ ZLayer.fromEffect(Ref.make(0)),
```

File: test-tests/shared/src/test/scala/zio/test/MutableRunnableSpecSpec.scala [hunk 2]
```diff
@@ -3,10 +3,10 @@ package zio.test
 import zio.test.Assertion._
 import zio.test.TestAspect._
 import zio.test.environment.TestEnvironment
-import zio.{Ref, ZIO}
+import zio.{Has, Ref, ZIO}
```
Comment (adamgfraser) [inline test-tests/shared/src/test/scala/zio/test/MutableRunnableSpecSpec.scala:6]:
```suggestion
import zio.{Has, Ref, ZIO, ZLayer}
```

**Comment by adamgfraser** [comment]
@sideeffffect Not flakiness but I think we are running into #4802. I suggested a couple of changes that will resolve the issue for now.

**Comment by sideeffffect** [comment]
Green :tada:

**Review by adamgfraser — APPROVED** [review]
@sideeffffect Thanks for your work on this!

**Pull request merged**